### ISSUE TITLE
Full baseline is never generated because since filter marks mutations as unchanged

### ISSUE THREAD

**Issue by rouke-broersma** [stryker-mutator/stryker-net#1803]
**Describe the bug**
The since filter marks mutations as unchanged based on git information, this causes the full baseline for the baseline feature to never be generated making the baseline feature useless.

**Expected behavior**
If a baseline does not exist, a full mutation report should be generated even if there were no changes based on git information.

**Additional context**
Probably a problem in the order of filter execution

**Comment by rouke-broersma** [comment]
This is a problem because the baseline filter implicitly enables the since filter. We should remove this implicit enable. However we then lose the capability of determining that the baseline is incorrect due to undetected changes..

@richardwerkman @Stijn-Rutten what do you think

**Comment by Stijn-Rutten** [comment]
Removing the implicit enable would not fix anything. When the since feature is disabled we would place all the results from the baseline in the current test run. Then we would not edit any  mutant results and thus none of the mutants covered by the baseline would change. Only new mutants would get a new result. All mutants covered by the old result could have wrong results. The most viable option I can think of right now is checking if a baseline is provided. If this is not the case we should disable both the baseline filter and the since filter. After we have done that we would run a complete test of all mutations. After that we can upload a fresh baseline.

**Comment by richardwerkman** [comment]
We could add a `create-baseline` command that stores the baseline in the desired storage provider after doing a complete run.

**Comment by rouke-broersma** [comment]
Wouldn't want to put the burden on the user when we can already detect that there is no baseline. This was supposed to be automatic (and used to be automatic).

**Comment by rouke-broersma** [comment]
The best solution IMO is to extract the actual git logic into a new class. Use this class both in the since filter and the baseline filter. Then we should not allow since and baseline to be enabled at the same time. This way both filters can share the existing logic but the baseline filter can choose not to enable the git filtering when there is no existing baseline.

**Comment by Stijn-Rutten** [comment]
@rouke-broersma would that mean either one is active at a given time? If baseline is enabled, since is disabled and vice versa?

**Comment by rouke-broersma** [comment]
Yes, at that point they cannot be enabled at the same time and there would also be no reason for the to be enabled at the same time. Since does since without the baseline stuff and baseline makes sure it takes git information into account to make sure the report is as correct as possible. Enabling both would then be pointless.

**Comment by richardwerkman** [comment]
Then it would make sense to turn them into commands like we wanted to. However that would introduce a breaking change

**Comment by rouke-broersma** [comment]
We can have them as both commands and options until the next major update and simply show a validation error if both are turned on for now.

**Comment by richardwerkman** [comment]
I guess the priority isn't that high as nobody outside of the stryker team noticed this (it seems). I stated on an implementation but got distracted and started on other work... Somebody else could pick this up if they want. Or I will continue after finishing the test detail feature

**Comment by richardwerkman** [comment]
Will be fixed by #1560

**Comment by StevenAngel-ai** [comment]
I am attempting to use Baseline, and running into this issue. Any update on when this might be fixed? I see the PR mentioned above has been open for going on three years now, but has not been merged. My team would love to be able to use this feature, but this issue does make it pretty much useless

**Comment by richardwerkman** [comment]
I completely forgot about this... There were some issues with the PR and we wanted to include it in a mayor update as it contained breaking changes if I remember correctly.

Maybe this is a good time to include this in a release, since dotnet 10 is releasing soon and we will have to update (breaking change) anyway! @rouke-broersma does that sound good to you?
